pulsar/src/main/java/com/hazelcast/jet/contrib/pulsar/PulsarSources.java

```diff
@@ -0,0 +1,374 @@
+     * is called "shared subscription" that allows multiple consumers to consume
+     * from the topic at the same time. The messages are sent round-robin to each
+     * connected consumer. Broker determines which consumer will receive a
+     * message from which topic partition. It does not require one-to-one
+     * mapping between partitions and consumers. Multiple consumers can get
+     * messages from same partition. With this source, the message ordering is
+     * not preserved.
+     * <p>
+     * Example usage:
+     * <pre>{@code
+     *
+     *   Map<String, Object> consumerConfig = new HashMap<>();
+     *   consumerConfig.put("consumerName", "hazelcast-jet-consumer");
+     *   consumerConfig.put("subscriptionName", "hazelcast-jet-subscription");
+     *   StreamSource<String> pulsarSource = PulsarSources.pulsarDistributed(
+     *          Collections.singletonList(topicName),
+     *          2,       // Preferred Local Parallelism
+     *          consumerConfig,
+     *          () -> PulsarClient.builder()
+     *                            .serviceUrl("pulsar://exampleserviceurl")
+     *                            .build(), // Client Supplier
+     *          () -> Schema.BYTES, // Schema Supplier Function
+     *          x -> new String(x.getData()) // Projection function that converts
+     *                                       // receiving bytes to String
+     *                                       // before emitting.
+     *          );
+     *  Pipeline pipeline = Pipeline.create();
+     *  StreamStage<Status> srcStage = p.readFrom(pulsarSource);
+     *
+     *  }</pre>
+     *
+     * @param topics the topics to consume, at least one is required
+     * @param preferredLocalParallelism the preferred number of local
+     *                                                    parallelism
+     * @param consumerConfig Pulsar consumer configurations that must
+     *                       contain consumer name, and subscription name.
+     * @param schemaSupplier supplies the schema for consuming messages
+     * @param projectionFn converts a Pulsar message to an emitted item.
+     * @param <M> the type of the message read by {@code PulsarConsumer}
+     * @param <T> the type of data emitted from {@code StreamSource}
+     * @return a stream source to use in {@link Pipeline#readFrom}
+     */
+    public static <M, T> StreamSource<T> pulsarConsumer(
+            @Nonnull List<String> topics,
+            @Nonnull int preferredLocalParallelism,
+            @Nonnull Map<String, Object> consumerConfig,
+            @Nonnull SupplierEx<PulsarClient> connectionSupplier,
+            @Nonnull SupplierEx<Schema<M>> schemaSupplier,
+            @Nonnull FunctionEx<Message<M>, T> projectionFn
+    ) {
+        checkSerializable(connectionSupplier, "connectionSupplier");
+        return SourceBuilder.timestampedStream("pulsar-distributed-stream-source", ctx -> new ConsumerContext(
```

Comment:
`new ConsumerContext(` -> `new ConsumerContext<>(`

Reply:
Changed this.

Reply:
fixed.